pkg/util/validation/limits.go

```diff
@@ -88,6 +96,13 @@ func (l *Limits) RegisterFlags(f *flag.FlagSet) {
 	f.IntVar(&l.MaxGlobalSeriesPerMetric, "ingester.max-global-series-per-metric", 0, "The maximum number of active series per metric name, across the cluster. 0 to disable.")
 	f.IntVar(&l.MinChunkLength, "ingester.min-chunk-length", 0, "Minimum number of samples in an idle chunk to flush it to the store. Use with care, if chunks are less than this size they will be discarded.")
 
+	f.IntVar(&l.MaxMetricsWithMetadataPerQuery, "ingester.max-metrics-with-metadata-per-query", 4000, "The maximum number of metrics with metadata that a query can return.")
+	f.IntVar(&l.MaxMetadataPerQuery, "ingester.max-metadata-per-query", 5, "The maximum number of metadata per metric that a query can return.")
```

Comment:
As discussed offline, I think limits in the read path are not necessary because the read metadata API endpoint allows to read all metadata only, so we never have write-path limits != read-path limits, thus write-path limits should be enough.

Reply:
Limits have been removed.